core/common/src/main/java/alluxio/resource/AlluxioResourceLeakDetector.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.resource;
+
+import io.netty.util.ResourceLeakDetector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An extension of the {@link ResourceLeakDetector} with custom error messages for use in the
+ * Alluxio codebase.
+ *
+ * @param <T> the type of resource the detector tracks
+ */
+public class AlluxioResourceLeakDetector<T> extends ResourceLeakDetector<T> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(AlluxioResourceLeakDetector.class);
+
+  /**
+   * Creates a new instance of the leak detector with the specific resource type and sampling
+   * interval.
+   *
+   * @param resourceType the resource class
+   * @param samplingInterval on average, how often a resource should be tracked
+   */
+  public AlluxioResourceLeakDetector(Class<?> resourceType, int samplingInterval) {
+    super(resourceType, samplingInterval);
+  }
+
+  @Override
+  protected void reportTracedLeak(String resourceType, String records) {
```

Comment:
For our better understanding, if you figure out how this traced/untraced leak differ, I think it makes sense to add to the comment here although it is `@Override`.

Reply:
Good point. I'll add some comments